ipaserver/plugins/cert.py

```diff
@@ -884,8 +886,8 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
                             "match requested principal") % gn.name)
             elif isinstance(gn, cryptography.x509.general_name.RFC822Name):
                 if principal_type == USER:
-                    if principal_obj and gn.value not in principal_obj.get(
```

Comment:
@felipevolpone sorry mate, one more tiny thing that I missed.  We need to retain the `if principal_obj and ...` because there is a scenario where `principal_obj = None` - namely, the PKINIT KDC certificate use case.  While it's unlikely that the CSR in this use case will include an RFC822Name, we should make the code resilient to this case.